eland/ml/exporters/es_gb_models.py

```diff
@@ -0,0 +1,434 @@
+
+    @property
+    def _definition(self) -> Mapping[Union[str, int], Any]:
+        return self._trained_model_result["trained_model_configs"][0]["definition"]
+
+    def _get_feature_names_in_(
+        self, preprocessors, feature_names, field_names
+    ) -> Tuple[List[str], Set[str]]:
+        input_field_names = set()
+        for preprocessor in preprocessors:
+            if "target_mean_encoding" in preprocessor:
+                feature_name = preprocessor["target_mean_encoding"]["feature_name"]
+                if feature_name in feature_names:
+                    input_field_names.add(preprocessor["target_mean_encoding"]["field"])
+            elif "frequency_encoding" in preprocessor:
+                feature_name = preprocessor["frequency_encoding"]["feature_name"]
+                if feature_name in feature_names:
+                    input_field_names.add(preprocessor["frequency_encoding"]["field"])
+
+            elif "one_hot_encoding" in preprocessor:
+                for feature_name in preprocessor["one_hot_encoding"][
+                    "hot_map"
+                ].values():
+                    if feature_name in feature_names:
+                        input_field_names.add(preprocessor["one_hot_encoding"]["field"])
+
+        for field_name in field_names:
+            if field_name in feature_names and field_name not in input_field_names:
+                input_field_names.add(field_name)
+
+        return feature_names, input_field_names
+
+    def fit(self, X, y, sample_weight=None, monitor=None) -> None:
+        """
+        Override of the sklearn fit() method. It does nothing since Elastic ML models are
+        trained in the Elastic Stack or imported.
+        """
+        # Do nothing, model if fitted using Elasticsearch API
+        pass
+
+    @property
+    @abstractmethod
+    def analysis_type(self):
+        """
+        Type of the data frame analysis. It can be classification or regression.
+        """
+        pass
+
+    @abstractmethod
+    def _initialize_init_(self) -> Union[DummyClassifier, DummyRegressor]:
+        pass
+
+
+class ESGradientBoostingClassifier(ESGradientBoostingModel, GradientBoostingClassifier):
+    """
+    Elastic ML model wrapper compatible with sklearn GradientBoostingClassifier.
+    """
+
+    def __init__(
+        self,
+        es_client: Union[str, List[str], Tuple[str, ...], "Elasticsearch"],
+        model_id: str,
+    ) -> None:
+        """
+        Parameters
+        ----------
+        es_client : Elasticsearch client argument(s)
+            - elasticsearch-py parameters or
+            - elasticsearch-py instance
+        model_id : str
+            The unique identifier of the trained inference model in Elasticsearch.
+
+        Raises
+        ------
+        NotImplementedError
+            Multi-class classification is not supported at the moment.
+        ValueError
+            The classifier should be defined for at least 2 classes.
```

Comment:
Can this also raise KeyError if names were to change.

Reply:
If the JSON schema of the trained model definition changes, it would also generate the KeyError. However, DFA is GA, so the schema is not supposed to change so quickly, and [Elasticsearch Python clients are only backward compatible](https://elasticsearch-py.readthedocs.io/en/v8.6.0/#compatibility) within the same major.

Reply:
I guess I wonder if we should also document that. But take your point it is kind of an exceptional exception.